src/main/java/bookstore/service/book/impl/BookServiceImpl.java

```diff
@@ -67,4 +69,13 @@ public List<BookDto> search(BookSearchParameters searchParameters) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> findAllByCategoryId(Long id) {
+        List<Book> books = bookRepository.findAllByCategoryId(id);
+        return books.stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
```

Comment:
Learn more about MapStruct. It can map List of BookDto to List of Book for you

Reply:
Try to do it using MapStruct